services/wfm/src/main/java/org/openkilda/wfm/topology/stats/FlowDirectionHelper.java

```diff
@@ -28,25 +27,30 @@ private FlowDirectionHelper() {}
         REVERSE
     }
 
-
     /**
      * Trys to determine the direction of the flow based on the cookie.
      *
-     * @param cookie cookie
+     * @param rawCookie cookie
      * @return flow direction
      */
-    public static Direction findDirection(long cookie) throws FlowCookieException {
-        // Kilda flow first number represents direction with 4 = forward and 2 = reverse
-        // high order nibble represents type of flow with a 2 representing a forward flow
-        // and a 4 representing the reverse flow
-        if (!isKildaCookie(cookie)) {
```

Comment:
Is this a legacy check? Do we really can remove it because there is no chance to catch a packet by "not kilda cookie" rule.

Reply:
No. But the only valid part of this check (mutually exclusive flags combination) I have moved into `Cookie` class. Other "weird" part about "next 20 bits must be zero" I have dropped because we do not have such "requirement".